### PR TITLE
refactor(compiler): remove unused exports from fs-watch-rebuild 

### PR DESCRIPTION
**Pull request by alicewriteswrongs**
This refactors some code in `src/compiler/fs-watch/fs-watch-rebuild.ts` to
DRY up a few things which were revealed by two unused exports in the
file.

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`npm run build`) was run locally and any changes were pushed
- [ ] Unit tests (`npm test`) were run locally and passed
- [ ] E2E Tests (`npm run test.karma.prod`) were run locally and passed
- [ ] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

GitHub Issue Number: N/A

## What is the new behavior?

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/compiler/fs-watch/fs-watch-rebuild.ts [hunk 1]
```diff
@@ -8,47 +8,85 @@ export const filesChanged = (buildCtx: d.BuildCtx) => {
...
+
+/**
+ * Script extensions which we want to be able to recognize
+ */
 const SCRIPT_EXT = ['ts', 'tsx', 'js', 'jsx'];
-export const isScriptExt = (ext: string) => SCRIPT_EXT.includes(ext);
+
+/**
+ * Helper to check if an extension is found in our list of script extensions
+ * @param ext a file extension
```
Comment (rwaskiewicz) [inline src/compiler/fs-watch/fs-watch-rebuild.ts:34]: Should this be something more like:
```suggestion
 * Helper to check if an extension is found in our list of script extensions
 * @param filePath the path of a file to check
```

I'm not entirely sure the summary on the first line aligns with what we're doing here. Thoughts?

File: src/compiler/fs-watch/fs-watch-rebuild.ts [hunk 2]
```diff
@@ -8,47 +8,85 @@ export const filesChanged = (buildCtx: d.BuildCtx) => {
...
+ */
+export const hasScriptExt = (filePath: string): boolean => SCRIPT_EXT.includes(getExt(filePath));
 
 const STYLE_EXT = ['css', 'scss', 'sass', 'pcss', 'styl', 'stylus', 'less'];
-export const isStyleExt = (ext: string) => STYLE_EXT.includes(ext);
 
-export const hasHtmlChanges = (config: d.Config, buildCtx: d.BuildCtx) => {
+/**
+ * Helper to check if an extension is found in our list of style extensions
+ * @param ext a file extension
```
Comment (rwaskiewicz) [inline src/compiler/fs-watch/fs-watch-rebuild.ts:43]: Should this be something more like:
```suggestion
 * Helper to check if an extension is found in our list of style extensions
 * @param filePath the path of a file to check
```
I'm not entirely sure the summary on the first line aligns with what we're doing here. Thoughts?

**Review by rwaskiewicz — CHANGES_REQUESTED** [review]
Two small asks to fix lint errors

**Comment by rwaskiewicz** [comment]
@alicewriteswrongs looks like lint (specifically Prettier) is failing for us for this PR

**Comment by alicewriteswrongs** [comment]
@rwaskiewicz should be updated now

**Review by rwaskiewicz — APPROVED** [review]
LGTM!

**Pull request merged**